File: typescript/Behavioral/19-State/State/DraftState.ts
import { BaseJobState } from "./BaseJobState";
import { AssignedState } from "./AssignedState";
import type { JobCard } from "../Context/JobCard";

export class DraftState extends BaseJobState {
  readonly name = "Draft";

  override assign(job: JobCard, handyman: string): void {
    job.assignee = handyman;
    job.transitionTo(new AssignedState());
    console.log(`  ✅ Assigned to ${handyman}`);
  }
}

File: csharp/Behavioral/19-State/State/DraftState.cs
namespace State;

public class DraftState : BaseJobState
{
    public override string Name => "Draft";

    public override void Assign(JobCard job, string handyman)
    {
        job.Assignee = handyman;
        job.TransitionTo(new AssignedState());
        Console.WriteLine($"  ✅ Assigned to {handyman}");
    }
}
